docs/Generators/SHA3.rst

```diff
@@ -0,0 +1,85 @@
+SHA3 RoCC Accelerator
+===================================
+
+Introduction
+-----------------------------------
+Secure hashing algorithms represent a class of hashing functions that provide four attributes: ease
+of hash computation, inability to generate the message from the hash (one-way property), inability
+to change the message and not the hash (weakly collision free property), and inability to find
+two messages with the same hash (strongly collision free property). The National Institute of
+Standards and Technology (NIST) recently held a competition for a new algorithm to be added to
+its set of Secure Hashing Algorithms (SHA). In 2012 the winner was determined to be the Keccak
+hashing function and a rough specification for SHA3 was established. The algorithm operates on
+variable length messages with a sponge function, and thus alternates between absorbing chunks of
+the message into a set of state bits and permuting the state. The absorbing is a simple bitwise
+XOR while the permutation is a more complex function composed of several operations, χ, θ, ρ,
+π, ι, that all perform various bitwise operations, including rotations, parity calculations, XORs,
+etc. The Keccak hashing function is parameterized for different sizes of state and message chunks
+but for this lab we will only support the Keccak-256 variant with 1600 bits of state and 1088 bit
+message chunks. In addition, for this lab we will ignore the variable length portion to avoid one
+of the most complicated parts of Keccak the padding. Our interface, which is discussed further
+below, assume a single chunk of message is ready to be absorbed and hashed. A diagram of the SHA3
```

Comment:
This may be true for the diagram but again the RTL supports multiple message chunks